pyanaconda/ui/tui/spokes/installation_source.py

```diff
@@ -144,85 +148,81 @@ def refresh(self, args=None):
         if self.data.method.method == "harddrive" and \
            get_mount_device(DRACUT_ISODIR) == get_mount_device(DRACUT_REPODIR):
             message = _("The installation source is in use by the installer and cannot be changed.")
-            self._window += [TextWidget(message), ""]
-            return True
+            self.window.add_with_separator(TextWidget(message))
+            return
 
         if args == 3:
             text = [TextWidget(_(p)) for p in self._protocols]
```

Comment:
Couldn't we replace `self._protocols` and `_methods` and add the items directly to the container with proper callbacks?